### PR TITLE
Add Console

### PR DESCRIPTION
**Pull request by vasilmkd**
I messed something up when rebasing and pushing. I didn't mean to close the previous PR.

#1197

### CODE REVIEW

File: concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+ *   def myProgram[F[_]: Sync]: F[Unit] =
+ *     for {
+ *       _ <- Console[F].println("Please enter your name: ")
+ *       n <- Console[F].readLine
+ *       _ <- if (n.nonEmpty) Console[F].println(s"Hello $n!")
+ *            else Console[F].printlnErr("Name is empty!")
+ *     } yield ()
+ * }}}
+ */
+class Console[F[_]] private (val F: Sync[F]) extends AnyVal {
```
Comment (djspiewak) [inline concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala]: Should be `final`

File: concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      loop()
+    }
+
+  /**
+   * Prints a value to the standard output using the implicit `cats.Show` instance.
+   *
+   * @param a value to be printed to the standard output
+   */
+  def print[A: Show](a: A): F[Unit] =
+    F.blocking(System.out.print(a.show))
```
Comment (djspiewak) [inline concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala]: Relatively minor niggle, but `a.show` should be run *outside* of the `blocking` pool.

```suggestion
  def print[A: Show](a: A): F[Unit] = {
    val text = a.show
    F.blocking(System.out.print(text))
  }
```

Same comment for the other `print`s.

File: concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+   */
+  def println[A: Show](a: A): F[Unit] =
+    F.blocking(System.out.println(a.show))
+
+  /**
+   * Prints a value to the standard error output using the implicit `cats.Show` instance.
+   *
+   * @param a value to be printed to the standard error output
+   */
+  def printErr[A: Show](a: A): F[Unit] =
```
Comment (djspiewak) [inline concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala]: Random idea: what if we called this `error` and the other `errorln`?

**Review by djspiewak — COMMENTED** [review]
More thoughts incoming as I get into the tests

File: concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala (hunk 2 above)
Comment (vasilmkd) [reply concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala]: Nice. Done.

File: concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala (hunk 1 above)
Comment (vasilmkd) [reply concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala]: Done.

File: concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala (hunk 3 above)
Comment (vasilmkd) [reply concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala]: I made the change.

File: core/shared/src/main/scala/cats/effect/IO.scala [hunk 4]
```diff
@@ -411,6 +412,23 @@ object IO extends IOCompanionPlatform with IOLowPriorityImplicits {
...
+  def readLine: IO[String] =
+    Console[IO].readLine
+
+  /**
+   * Prints a value to the standard output followed by a new line using the implicit `cats.Show` instance.
+   *
+   * @param a value to be printed to the standard output
+   */
+  def println[A: Show](a: A): IO[Unit] =
+    Console[IO].println(a)
```
Comment (djspiewak) [inline core/shared/src/main/scala/cats/effect/IO.scala]: I think we should probably add the other forwarders as well: `print`, `error`, and `errorln`.

Comment (vasilmkd) [reply core/shared/src/main/scala/cats/effect/IO.scala]: I only added `IO.print`. `IO.error` might be really confusing to users. I also added scaladoc that points to the full `Console`.

**Pull request merged**